moc_openstack/templates/surnfq.erb

```diff
@@ -6,79 +6,51 @@ net4=<%= scope.function_hiera(['quickstack::params::surnet4']) %>
-iptables -L $QNAME -nv --line-numbers | grep $net4 > /dev/null || iptables -I $QNAME -s $net4 -j ACCEPT
-iptables -L $QNAME -nv --line-numbers | grep $net3 > /dev/null || iptables -I $QNAME -s $net3 -j ACCEPT
-iptables -L $QNAME -nv --line-numbers | grep $net2 > /dev/null || iptables -I $QNAME -s $net2 -j ACCEPT
-iptables -L $QNAME -nv --line-numbers | grep $net1 > /dev/null || iptables -I $QNAME -s $net1 -j ACCEPT
+iptables -L -n | grep suricata| grep Chain > /dev/null || iptables -N suricata
+iptables -L $QNAME -n | grep suricata > /dev/null || iptables -I $QNAME -j suricata
 
-lnum=`iptables -L $QNAME -nv --line-numbers | grep $net1 | cut -f 1 -d " "|xargs -n1|sort -r -n`
+lnum=`iptables -L $QNAME -nv --line-numbers | grep suricata | cut -f 1 -d " "|xargs -n1|sort -r -n`
 if [ "$lnum" != "1" ]; then
     for line in $lnum; do
         iptables -D $QNAME $line
     done
-    iptables -I $QNAME 1 -s $net1 -j ACCEPT
-fi
-
-lnum=`iptables -L $QNAME -nv --line-numbers | grep $net2 | cut -f 1 -d " "|xargs -n1|sort -r -n`
-if [ "$lnum" != "2" ]; then
-    for line in $lnum; do
-        iptables -D $QNAME $line
-    done
-    iptables -I $QNAME 2 -s $net2 -j ACCEPT
-fi
-
-lnum=`iptables -L $QNAME -nv --line-numbers | grep $net3 | cut -f 1 -d " "|xargs -n1|sort -r -n`
-if [ "$lnum" != "3" ]; then
-    for line in $lnum; do
-        iptables -D $QNAME $line
-    done
-    iptables -I $QNAME 3 -s $net3 -j ACCEPT
-fi
-
-lnum=`iptables -L $QNAME -nv --line-numbers | grep $net4 | cut -f 1 -d " "|xargs -n1|sort -r -n`
-if [ "$lnum" != "4" ]; then
-    for line in $lnum; do
-        iptables -D $QNAME $line
-    done
-    iptables -I $QNAME 4 -s $net4 -j ACCEPT
-fi
-
-lnum=`iptables -L $QNAME -nv --line-numbers | grep $net5 | cut -f 1 -d " "|xargs -n1|sort -r -n`
-if [ "$lnum" != "5" ]; then
-    for line in $lnum; do
-        iptables -D $QNAME $line
-    done
-    iptables -I $QNAME 5 -s $net5 -j ACCEPT
-fi
-
-lnum=`iptables -L $QNAME -nv --line-numbers | grep $net6 | cut -f 1 -d " "|xargs -n1|sort -r -n`
-if [ "$lnum" != "6" ]; then
-    for line in $lnum; do
-        iptables -D $QNAME $line
-    done
-    iptables -I $QNAME 6 -s $net6 -j ACCEPT
-fi
-
-lnum=`iptables -L $QNAME -nv --line-numbers | grep $net7 | cut -f 1 -d " "|xargs -n1|sort -r -n`
-if [ "$lnum" != "7" ]; then
-    for line in $lnum; do
-        iptables -D $QNAME $line
-    done
-    iptables -I $QNAME 7 -s $net7 -j ACCEPT
+    iptables -I $QNAME -j suricata
 fi
 
 lnum=`iptables -L $QNAME -nv --line-numbers | grep NFQUEUE | cut -f 1 -d " "|xargs -n1|sort -r -n`
-if [ "$lnum" != "8" ]; then
-    for line in $lnum; do
-        iptables -D $QNAME $line
-    done
-    iptables -I $QNAME 8 -m mark ! --mark 0x1/0x1 -j NFQUEUE
-fi
-
+if [ "$lnum" != "" ]; then
+    for line in $lnum; do
+        iptables -D $QNAME $line
+    done
+fi
+
+
+runningsum=`iptables -L suricata -n| md5sum`
+#echo $runningsum
+##echo $checksum
+#
+#[ "$checksum" == "$runningsum" ] && exit
+#
```

Comment:
I am not sure if this is a good way. Since we have multiple environments, plus 'n' number of staging environments, the value of checksum will change everytime. This checksum value depends on the $net1, $net2, $net3,..... used. We will see more issues with this since some environments will have less networks and some will have more based on the hiera file and this will cause issues.

This is adding additional dependency of knowing that suricata may go wrong everytime.